### PR TITLE
refactor: follow Rails conventions all over the place

### PR DESCRIPTION
**Pull request by rainerborene**
Related to #109  

ping @joelhawksley

### CODE REVIEW

File: lib/action_view/component/conversion.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+module ActionView
+  module Component # :nodoc:
```
Comment (juanmanuelramallo) [inline lib/action_view/component/conversion.rb]:
```suggestion
  module ActiveModel # :nodoc:
```
And previous line should be removed. This is a monkey patch from #129

Comment (rainerborene) [reply lib/action_view/component/conversion.rb]: It's already there on the [last line](https://github.com/github/actionview-component/pull/138/files#diff-7348f38d58d7fe263398878a3236faa2R21) of `component.rb` file!

Comment (juanmanuelramallo) [reply lib/action_view/component/conversion.rb]: Ahh got it! Nice one

File: README.md [hunk 2]
```diff
@@ -225,7 +223,7 @@ In general, we’ve found it makes the most sense to test components based on th
 To test a specific variant you can wrap your test with the `with_variant` helper method as:
 
 ```ruby
-def test_render_component_for_tablet
+test "render component for table" do
```
Comment (joelhawksley) [inline README.md]:
```suggestion
test "render component for tablet" do
```

File: README.md [hunk 3]
```diff
@@ -204,12 +204,10 @@ An error will be raised:
...
 ```ruby
-require "action_view/component/test_helpers"
-
-class MyComponentTest < Minitest::Test
-  include ActionView::Component::TestHelpers
+require "action_view/component/test_case"
 
-  def test_render_component
+class MyComponentTest < ActionView::Component::TestCase
+  test "render component" do
```
Comment (joelhawksley) [inline README.md]: Is using strings for test names convention these days? The Rails tests I've looked at recently used method names.

File: README.md [hunk 4]
```diff
@@ -32,7 +32,7 @@ $ bundle
 In `config/application.rb`, add:
 
 ```bash
-require "action_view/component"
+require "action_view/component/railtie"
```
Comment (joelhawksley) [inline README.md]: Note: we should do a minor version bump with this change.

**Comment by joelhawksley** [comment]
@rainerborene I'd love to merge this patch. Might you be able to make the minor suggested changes?

**Comment by rainerborene** [comment]
Sure. I will apply these changes until the end of this week.

File: README.md (hunk 2 above)
Comment (rainerborene) [reply README.md]: Well, that's debatable. If you take a look on the official documentation, you'll see some examples using this syntax.

File: README.md (hunk 3 above)
Comment (rainerborene) [reply README.md]: Well, that's debatable. If you take a look on the official documentation, you'll see some examples using this syntax. By the way, which convention GitHub adopted internally? 🤔

File: README.md (hunk 4 above)
Comment (rainerborene) [reply README.md]: Let's keep the require `railtie` naming convetion on the README file. To keep backward compatibility, the older require `railtie` statement was included on `component.rb` file (see the last commit).

**Comment by rainerborene** [comment]
@joelhawksley done!

File: lib/action_view/component.rb [hunk 5]
```diff
@@ -1,6 +1,22 @@
 # frozen_string_literal: true
 
-require "action_view/component/monkey_patch"
-require "action_view/component/active_model_conversion_monkey_patch"
-require "action_view/component/base"
+require "active_model"
+require "action_view"
+require "active_support/dependencies/autoload"
 require "action_view/component/railtie"
```
Comment (joelhawksley) [inline lib/action_view/component.rb]: Should we plan to deprecate this line?

Comment (rainerborene) [reply lib/action_view/component.rb]: I think we should. 👍

Comment (joelhawksley) [reply lib/action_view/component.rb]: Ok! Can you file an issue?

**Pull request merged**